### PR TITLE
Book1 fig 2.12

### PR DESCRIPTION
**Pull request by TakshPanchal**
### Figure Number

**Book 1 Fig 2.12**

### Figures

| Before PR | After PR |
| :-: | :-: |
|image|Image

### Issue 

#4  

## Steps
- [x] Latexified the code
- [x] Jaxified the code
- [x] Added interactive demo by the varying value of temperature (T)

## Checklist:

- [X] Performed a self-review of the code
- [X] Tested on Google Colab.

### CODE REVIEW

**Comment by karm-patel** [comment]
Hi @TakshPanchal 
## Figure related comments:
1. Do we need the right and top spines of each figure? We can remove them using `sns.despine()`. please observe similar figures in closed PR.
2. Currently labels of `x_ticks` are missing and it's difficult to guess the value of logits without seeing the caption, can you figure out what are those labels?
3. Since all 3 subplots have same `y_lim` you can show only `y_ticks` for only first plot and hide for other two.
4. I think it would be good to rename `y_label` as "S(a|T)"  (In latex format) and `x_label` as "logits (a)". Refer equation 2.95.

File: notebooks/book1/02/softmax_plot.ipynb [hunk 1]
```diff
@@ -0,0 +1,185 @@
+{
```
Comment (karm-patel) [inline notebooks/book1/02/softmax_plot.ipynb:1]: <p>Add title of notebook, observe other notebooks</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (karm-patel) [inline notebooks/book1/02/softmax_plot.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        plot_softmax_distribution(T, a, f"softmax_dist_{i+1}_latexified")
</pre><p>Now you can remove <code>_latexified</code>, <code>save_fig</code> will add it.   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/59/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Comment by karm-patel** [comment]
## IMP Comments
1. Can I know why you have moved from `sub_plots` to individual plots? This will create one problem, think of it has we have latex source code of this figure. And since in original code we are saving only 1 figure (which contains 3 plots), latex source code also called only one figure. Now you have added 3 individual figure so now author of this book needs to change whole latex source code and find out 3 pdfs of figure which you generated.   So essence is we should not change the things which results to change in latex source code for example figure names.

File: notebooks/book1/02/softmax_plot.ipynb (hunk 1 above)
Comment (TakshPanchal) [reply notebooks/book1/02/softmax_plot.ipynb:1]: okay. understood.

**Comment by karm-patel** [comment]
Hi @TakshPanchal, This looks good to me, you can create a PR on the main repo, but before that refer these instructions: https://github.com/probml/pyprobml/blob/master/notebooks/README.md

Also, change the caption while putting the After PR plot.

**Comment by TakshPanchal** [comment]
How to get the original caption?

**Comment by karm-patel** [comment]
You can just copy-paste the caption from the book without worrying about latex symbols.
